File: fl_umeng_apm/ios/Classes/UMengAPMPlugin.m
#import "UMengAPMPlugin.h"
#import <UMAPM/UMCrashConfigure.h>
#import <UMAPM/UMAPMConfig.h>

@implementation UMengAPMPlugin

+ (void)registerWithRegistrar:(NSObject <FlutterPluginRegistrar> *)registrar {
    FlutterMethodChannel *channel = [FlutterMethodChannel
            methodChannelWithName:@"UMeng.apm"
                  binaryMessenger:registrar.messenger];

    [channel setMethodCallHandler:^(FlutterMethodCall *_Nonnull call, FlutterResult _Nonnull result) {
        if ([@"init" isEqualToString:call.method]) {
            NSDictionary *crash = call.arguments;
            [NSURLProtocol registerClass:[NSURLProtocol class]];
            [UMCrashConfigure enableNetworkForProtocol:[crash[@"enableNetworkForProtocol"] boolValue]];
            UMAPMConfig *config = [UMAPMConfig defaultConfig];
            config.crashAndBlockMonitorEnable = [crash[@"enableCrashAndBlock"] boolValue];
            config.launchMonitorEnable = [crash[@"enableLaunch"] boolValue];
            config.memMonitorEnable = [crash[@"enableMEM"] boolValue];
            config.oomMonitorEnable = [crash[@"enableOOM"] boolValue];
            config.networkEnable = [crash[@"networkEnable"] boolValue];
            [UMCrashConfigure setAPMConfig:config];
            result(@(YES));
        } else if ([@"setAppVersion" isEqualToString:call.method]) {
            NSDictionary *args = call.arguments;
            [UMCrashConfigure setAppVersion:args[@"version"] buildVersion:args[@"buildId"]];
            result(@(YES));
        } else {
            result(FlutterMethodNotImplemented);
        }
    }];
}
@end

File: fl_umeng_apm/ios/Classes/UMengAPMPlugin.swift
// import Flutter
//
// public class UMengAPMPlugin: NSObject, FlutterPlugin {
//     private var channel: FlutterMethodChannel
//
//     public static func register(with registrar: FlutterPluginRegistrar) {
//         let channel = FlutterMethodChannel(name: "UMeng.apm", binaryMessenger: registrar.messenger())
//         let plugin = UMengAPMPlugin(channel)
//         registrar.addMethodCallDelegate(plugin, channel: channel)
//     }
//
//     init(_ channel: FlutterMethodChannel) {
//         self.channel = channel
//     }
//
//     public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
//         switch call.method {
//         case "init":
//             let args = call.arguments as! [String: Any]
//             UMCrashConfigure.enableNetwork(forProtocol: args["enableNetworkForProtocol"] as! Bool)
//             let config = UMAPMConfig.default()
//             config.crashAndBlockMonitorEnable = args["enableCrashAndBlock"] as! Bool
//             config.launchMonitorEnable = args["enableLaunch"] as! Bool
//             config.memMonitorEnable = args["enableMEM"] as! Bool
//             config.oomMonitorEnable = args["enableOOM"] as! Bool
//             config.networkEnable = args["networkEnable"] as! Bool
//             UMCrashConfigure.setAPMConfig(config)
//             result(true)
//         case "setAppVersion":
//             let args = call.arguments as! [String: Any]
//             let version = args["version"] as! String
//             let buildId = args["buildId"] as! String
//             UMCrashConfigure.setAppVersion(version, buildVersion: buildId)
//             result(true)
//         default:
//             result(FlutterMethodNotImplemented)
//         }
//     }
// }
